code/datums/quirks/positive_quirks/chipped.dm

```diff
@@ -14,44 +14,70 @@
 	)
 	/// Variable that holds the chip, used on removal.
 	var/obj/item/skillchip/installed_chip
-	var/datum/callback/itchy_timer
+	///itchy status effect we give our owner
+	var/datum/itchy_effect
 
 /datum/quirk_constant_data/chipped
 	associated_typepath = /datum/quirk/chipped
 	customization_options = list(/datum/preference/choiced/chipped)
 
 /datum/quirk/chipped/add_to_holder(mob/living/new_holder, quirk_transfer, client/client_source)
-	var/obj/item/skillchip/chip_pref = GLOB.quirk_chipped_choice[client_source?.prefs?.read_preference(/datum/preference/choiced/chipped)]
+	var/chip_pref = client_source?.prefs?.read_preference(/datum/preference/choiced/chipped)
 
-	if(!chip_pref)
+	if(isnull(chip_pref))
 		return ..()
-
-	gain_text = span_notice("The [chip_pref] in your head itches a bit.")
+	installed_chip = GLOB.quirk_chipped_choice[chip_pref] || GLOB.quirk_chipped_choice[pick(GLOB.quirk_chipped_choice)]
+	gain_text = span_notice("The [installed_chip::name] in your head itches a bit.")
 	lose_text = span_notice("Your head stops itching so much.")
 	return ..()
 
 /datum/quirk/chipped/add_unique(client/client_source)
-
-	var/preferred_chip = GLOB.quirk_chipped_choice[client_source?.prefs?.read_preference(/datum/preference/choiced/chipped)]
-	if(isnull(preferred_chip))  //Client is gone or they chose a random chip
-		preferred_chip = GLOB.quirk_chipped_choice[pick(GLOB.quirk_chipped_choice)]
+	if(!iscarbon(quirk_holder))
+		return
 
 	var/mob/living/carbon/quirk_holder_carbon = quirk_holder
-	if(iscarbon(quirk_holder))
-		installed_chip = new preferred_chip()
-		quirk_holder_carbon.implant_skillchip(installed_chip, force = TRUE)
+	installed_chip = new installed_chip()
+
+	RegisterSignals(installed_chip, list(COMSIG_QDELETING, COMSIG_SKILLCHIP_REMOVED), PROC_REF(remove_effect))
+	RegisterSignal(installed_chip, COMSIG_SKILLCHIP_IMPLANTED, PROC_REF(apply_effect))
+
+	quirk_holder_carbon.implant_skillchip(installed_chip, force = TRUE)
 	installed_chip.try_activate_skillchip(silent = FALSE, force = TRUE)
 
-	var/obj/item/organ/brain/itchy_brain = quirk_holder.get_organ_by_type(ORGAN_SLOT_BRAIN)
-	itchy_timer = addtimer(CALLBACK(src, PROC_REF(cause_itchy), itchy_brain), rand(5 SECONDS, 10 MINUTES)) // they get The Itch from a poor quality install every so often
+/datum/quirk/chipped/proc/apply_effect(datum/source, obj/item/brain_applied)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/quirk_holder_carbon = quirk_holder
+	if(brain_applied == quirk_holder_carbon.get_organ_slot(ORGAN_SLOT_BRAIN))
+		itchy_effect = quirk_holder.apply_status_effect(/datum/status_effect/itchy_skillchip_quirk)
+
+/datum/quirk/chipped/proc/remove_effect(datum/source, obj/item/brain_removed)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/quirk_holder_carbon = quirk_holder
+	if(QDELING(source) || brain_removed == quirk_holder_carbon.get_organ_slot(ORGAN_SLOT_BRAIN))
+		quirk_holder.remove_status_effect(itchy_effect)
+		itchy_effect = null
```

Comment:
```suggestion
		QDEL_NULL(itchy_effect)
```

Reply:
Though tbh I don't know why you bother keeping a ref to the status effect rather than just do `apply_status_effect(/datum/status_effect/itchy_skillchip_quirk)` and `remove_status_effect(/datum/status_effect/itchy_skillchip_quirk)`